libvmi/driver/xen/xen.c

```diff
@@ -2455,15 +2455,51 @@ xen_set_vcpureg(
 #if defined(ARM32) || defined(ARM64)
     return xen_set_vcpureg_arm(vmi, value, reg, vcpu);
 #elif defined(I386) || defined (X86_64)
-    if (!xen_get_instance(vmi)->hvm) {
-        if (8 == xen_get_instance(vmi)->addr_width) {
-            return xen_set_vcpureg_pv64(vmi, value, reg, vcpu);
-        } else {
-            return xen_set_vcpureg_pv32(vmi, value, reg, vcpu);
+    if (xen_get_instance(vmi)->hvm) {
+        switch (reg) {
+        case RAX:
+        case RBX:
+        case RCX:
+        case RDX:
+        case RBP:
+        case RSI:
+        case RDI:
+        case RSP:
+        case R8:
+        case R9:
+        case R10:
+        case R11:
+        case R12:
+        case R13:
+        case R14:
+        case R15:
+        case RIP:
+        case RFLAGS:
+        case CR0:
+        case CR2:
+        case CR3:
```

Comment:
I just updated the code to remove CR3 support.   It seems like PV doesn't manipulate any of the other registers so everything should be equivalent to the VCPU-safe HVM methods